main.js

```diff
@@ -71,22 +81,58 @@ app.on('ready', function () {
     }
   };
 
+  const promiseTimeout = function (ms, promise) {
+
+    // Create a promise that rejects in <ms> milliseconds
+    let timeout = new Promise((resolve, reject) => {
+      let id = setTimeout(() => {
+        clearTimeout(id);
+        reject('Timed out in ' + ms + 'ms.')
+      }, ms)
+    })
+
+    // Returns a race between our timeout and the passed in promise
+    return Promise.race([
+      promise,
+      timeout
+    ])
+  }
+
   ipcMain.on(constants.IPC_CONFIGURE_ACCOUNT, (event, googleConfig, googleOptions) => {
     this.googleConfig = googleConfig;
     this.googleOptions = googleOptions;
   });
 
-  ipcMain.on(constants.IPC_GOOGLE_AUTH, (event) => {
-    const myApiOauth = electronOauth2(this.googleConfig, windowParams);
-    myApiOauth.getAccessToken(this.googleOptions)
-      .then(token => {
+  ipcMain.on(constants.IPC_GOOGLE_AUTH, (event, googleConfig, googleOptions) => {
+    const myApiOauth = electronOauth2(googleConfig, windowParams);
+
+    let doLogin = function () {
+      return myApiOauth.getAccessToken(googleOptions);
+    };
+
+    // Apply a timeout of 15 seconds to doLogin
+    let doIt = promiseTimeout(150000, doLogin());
```

Comment:
In practice I found that it would close the app even if I hadn't typed anything in the login dialog. Is that expected? I would expect the timeout to occur after the user attempts to log in, not if they launch it and don't type anything yet.

Reply:
Humm, ok. This was added to avoid getting stuck when no internet is available. Otherwise, the application only could be closed killing the process, clearly a non option for average users. 

I would create a separated bug story to address this wrong behavior.

Reply:
Improvement story created : https://broadinstitute.atlassian.net/browse/DSDEGP-2022